pkg/cloudprovider/providers/vsphere/vclib/diskmanagers/vmdm.go

```diff
@@ -20,18 +20,21 @@ type vmDiskManager struct {
 // Create implements Disk's Create interface
 // Contains implementation of VM based Provisioning to provision disk with SPBM Policy or VSANStorageProfileData
 func (vmdisk vmDiskManager) Create(ctx context.Context, datastore *vclib.Datastore) (err error) {
-	if vmdisk.volumeOptions.StoragePolicyName != "" {
-		pbmClient, err := vclib.NewPbmClient(ctx, datastore.Client())
-		if err != nil {
-			glog.Errorf("Error Occurred while creating new pbmClient, err: %+v", err)
-			return err
-		}
+
+	pbmClient, err := vclib.NewPbmClient(ctx, datastore.Client())
+	if err != nil {
+		glog.Errorf("Error Occurred while creating new pbmClient, err: %+v", err)
+		return err
+	}
+
+	if vmdisk.volumeOptions.StoragePolicyID == "" && vmdisk.volumeOptions.StoragePolicyName != "" {
```

Comment:
Only StoragePolicyName check should be fine. This should be just fine -
 vmdisk.volumeOptions.StoragePolicyName != ""